File: cpp/src/observatory/phase/FillReceiveQueuePhase.cpp
#include "FillReceiveQueuePhase.h"
#include <observatory/Benchmark.h>

#include <utility>
#include <observatory/operation/Operation.h>

namespace Observatory {

FillReceiveQueuePhase::FillReceiveQueuePhase(Benchmark &benchmark, const Operation &operation) :
        BenchmarkPhase(benchmark),
        fillReceiveQueue(operation.needsFilledReceiveQueue()) {}

const char* FillReceiveQueuePhase::getName() {
    return "FillReceiveQueuePhase";
}

Status FillReceiveQueuePhase::execute() {
    if(fillReceiveQueue) {
        return getBenchmark().fillReceiveQueue();
    }

    return Status::OK;
}

}

File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/FillReceiveQueuePhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

class FillReceiveQueuePhase extends BenchmarkPhase {

    private final boolean fillReceiveQueue;

    FillReceiveQueuePhase(Connection connection, Operation operation) {
        super(connection);

        this.fillReceiveQueue = operation.needsFilledReceiveQueue();
    }

    @Override
    Status execute() {
        if (fillReceiveQueue) {
            return getConnection().fillReceiveQueue();
        }

        return Status.OK;
    }
}
